File: output/llama_4_maverick/avatar/algo_based_translation/java/python/codeforces_299_A.py

def main():
    n = int(input())
    a = list(map(int, input().split()))
    a.sort()
    min_val = a[0]
    for value in a:
        if value % min_val != 0:
            print(-1)
            return
    print(min_val)

if __name__ == "__main__":
    main()

File: output/deepseek_chat/avatar/direct_translation/python/java/codeforces_299_A.java
import java.util.*;
import java.io.*;

public class codeforces_299_A {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] input = br.readLine().split(" ");
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(input[i]);
        }
        Arrays.sort(a);
        boolean flag = false;
        for (int x : a) {
            if (x % a[0] != 0) {
                flag = true;
                break;
            }
        }
        System.out.println(flag ? -1 : a[0]);
    }
}
